sync/handlers/leafs_request_test.go

```diff
@@ -29,13 +30,17 @@ func TestLeafsRequestHandler_OnLeafsRequest(t *testing.T) {
 	memdb := rawdb.NewMemoryDatabase()
 	trieDB := trie.NewDatabase(memdb)
 
-	corruptedTrieRoot, _, _ := trie.GenerateTrie(t, trieDB, 100, common.HashLength)
+	corruptedTrieRoot, _, _ := syncutils.GenerateTrie(t, trieDB, 100, common.HashLength)
+	tr, err := trie.New(trie.TrieID(corruptedTrieRoot), trieDB)
```

Comment:
https://github.com/ava-labs/coreth/pull/361/files#diff-3de72fe1927b71bef9ec9ba4ecb486c6aec27ef95963883e6c21bf7aad06fbbbL106 doesn't work anymore since there's no access to `.diskdb` so I take a trie and the underlying diskdb as arguments (instead of diskdb and a triedb and a root)